domain/domain.go

```diff
@@ -1221,6 +1245,128 @@ func (do *Domain) NotifyUpdatePrivilege(ctx sessionctx.Context) {
 	}
 }
 
+// ServerID gets serverID.
+func (do *Domain) ServerID() uint64 {
+	return atomic.LoadUint64(&do.serverID)
+}
+
+func (do *Domain) getServerIDSession() *concurrency.Session {
+	s, _ := do.serverIDSession.Load().(*concurrency.Session)
+	return s
+}
+
+const (
+	serverIDEtcdPath             = "/tidb/server_id"
+	acquireServerIDRetryCnt      = 100
+	acquireServerIDRetryInterval = 300 * time.Millisecond
+	acquireServerIDTimeout       = 3 * time.Second
+)
+
+func (do *Domain) acquireServerID(ctx context.Context) error {
+	cfg := config.GetGlobalConfig()
+
+	atomic.StoreUint64(&do.serverID, 0)
```

Comment:
Could the new connection be forbidden to create if tidb-server cannot acquire a server ID or `isLostConnectionToPD` is true?
So we can drop the compatibility support of `compatible-kill-query`.